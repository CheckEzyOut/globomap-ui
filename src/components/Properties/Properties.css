/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.item-properties {
  box-sizing: border-box;
}

.item-properties .item-prop {
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  margin-bottom: 8px;
}

.item-properties .item-prop > span,
.item-properties .item-prop > span:first-child {
  font-size: 12px;
  padding: 4px 25px;
}

.item-properties .item-prop > span:first-child {
  font-weight: bold;
  font-size: 14px;
  color: #444;
  font-family: 'opensans-regular', Arial, Helvetica, sans-serif;
}

.item-properties .item-prop a {
  text-decoration: none;
}

.item-properties .item-prop .icon {
  display: inline-block;
  margin-right: 2px;
  width: 5px;
}

.item-properties .btn-show-more {
  margin: 8px 0;
  background-color: #efefef;
}

.item-properties .l-0 .sub-item-prop { background-color: #fffddd; }
.item-properties .l-1 .sub-item-prop { background-color: #f8f6cb; }
.item-properties .l-2 .sub-item-prop { background-color: #efedb8; }
