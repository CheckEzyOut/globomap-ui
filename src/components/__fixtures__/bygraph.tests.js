export const items = {
  "graph": "acl",
  "edges": [],
  "nodes": [
    {
      "_id": "tag_firewall/gproxy_LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
      "id": "LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
      "name": ".s3-sa-east-1.amazonaws.com",
      "provider": "gproxy",
      "timestamp": 1512025259,
      "properties": null,
      "properties_metadata": null,
      "type": "tag_firewall"
    }
  ],
  "subnodes": []
};

export const graphs = [
  {
    "name": "acl",
    "colorClass": "graph-color0",
    "enabled": false
  },
  {
    "name": "cloud_stack",
    "colorClass": "graph-color1",
    "enabled": false
  },
  {
    "name": "custeio",
    "colorClass": "graph-color2",
    "enabled": false
  },
  {
    "name": "dns",
    "colorClass": "graph-color3",
    "enabled": false
  },
  {
    "name": "galeb",
    "colorClass": "graph-color4",
    "enabled": false
  },
  {
    "name": "load_balancing",
    "colorClass": "graph-color5",
    "enabled": false
  },
  {
    "name": "networking_topology",
    "colorClass": "graph-color6",
    "enabled": false
  },
  {
    "name": "permission",
    "colorClass": "graph-color7",
    "enabled": false
  },
  {
    "name": "tsuru",
    "colorClass": "graph-color8",
    "enabled": false
  },
  {
    "name": "zabbix",
    "colorClass": "graph-color9",
    "enabled": false
  }
];
