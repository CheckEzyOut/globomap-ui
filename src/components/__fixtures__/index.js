export const node = {
  "_id": "comp_unit/globomap_5d9ed63b-1985-4d3a-98d4-8b0195ce1c53",
  "id": "5d9ed63b-1985-4d3a-98d4-8b0195ce1c53",
  "name": "cartolaapi-02-145989229527",
  "provider": "globomap",
  "timestamp": 1511441091,
  "properties": {
    "equipment_type": "Servidor Virtual",
    "host": "cmal18mp11lc03",
    "ips": [
      "10.132.67.64"
    ],
    "uuid": "5d9ed63b-1985-4d3a-98d4-8b0195ce1c53"
  },
  "properties_metadata": {
    "equipment_type": {
      "description": "Equipment Type"
    },
    "host": {
      "description": "Host name"
    },
    "ips": {
      "description": "IPs"
    },
    "uuid": {
      "description": "UUID"
    }
  },
  "type": "comp_unit",
  "edges": {
    "in": [
      {
        "_id": "access/aclapi_251406",
        "_from": "network/napi_v4_36200",
        "_to": "comp_unit/globomap_5d9ed63b-1985-4d3a-98d4-8b0195ce1c53",
        "id": "251406",
        "name": "10.130.7.224/27-10.132.67.64/32",
        "provider": "aclapi",
        "timestamp": 1512019453,
        "properties": {
          "protocol": "tcp",
          "destination": "10.132.67.64/32",
          "source": "10.130.7.224/27",
          "action": "permit"
        },
        "properties_metadata": {
          "protocol": {
            "description": "Protocol of access"
          },
          "destination": {
            "description": "Destination address"
          },
          "source": {
            "description": "Source address"
          },
          "action": {
            "description": "Access action"
          }
        },
        "type": "access",
        "dir": "in"
      }
    ],
    "out": [],
    "graph": "acl"
  },
  "uuid": "2dc992f7-f9fd-432a-a9d6-b0484d4861b3",
  "items": [],
  "exist": true
};

export const nodes = [
  {
    "_id": "tag_firewall/gproxy_LmdpdGh1Yi1jbG91ZC5zMy5hbWF6b25hd3MuY29t",
    "id": "LmdpdGh1Yi1jbG91ZC5zMy5hbWF6b25hd3MuY29t",
    "name": ".github-cloud.s3.amazonaws.com",
    "provider": "gproxy",
    "timestamp": 1511247666,
    "properties": null,
    "properties_metadata": null,
    "type": "tag_firewall"
  },{
    "_id": "tag_firewall/gproxy_LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
    "id": "LnMzLXNhLWVhc3QtMS5hbWF6b25hd3MuY29t",
    "name": ".s3-sa-east-1.amazonaws.com",
    "provider": "gproxy",
    "timestamp": 1511247655,
    "properties": null,
    "properties_metadata": null,
    "type": "tag_firewall"
  }
];
