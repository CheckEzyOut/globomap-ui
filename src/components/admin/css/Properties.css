/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.key {
  margin-right: 5px;
}

.key-value {
  display: flex;
  font-size: 0.875rem;
  margin-bottom: 5px;
}

.data div:nth-last-child(2) {
  border-bottom: 1px solid #ccc;
  margin-bottom: 18px;
  padding-bottom: 30px;
}

.add {
  text-align: right;
}

.add-button {
  background-color: transparent;
  box-shadow: none;
  color: #0669de;
  font-size: 12px;
  font-weight: 700;

  border: 0;
  border-radius: 4px;
  box-sizing: border-box;
  cursor: pointer;
  text-decoration: none;
  text-transform: uppercase;
}

.add-button svg, .remove-button svg {
  height: 14px;
  vertical-align: top;
  width: 16px;
}

.add-button svg path {
  fill: #0769de;
}
