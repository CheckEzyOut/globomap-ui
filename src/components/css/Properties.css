.item-properties {
  padding: 8px 0;
}

.item-properties table {
  width: 100%;
  border-spacing: 0;
}

.item-properties th {
  text-align: right;
  width: 36%;
  vertical-align: top;
}

.item-properties th,
.item-properties td {
  font-size: 12px;
  padding: 4px 8px;
}

.item-properties td span {
  display: block;
}

.item-properties .btn-show-more {
  margin: 8px 0;
  background-color: #efefef;
}
