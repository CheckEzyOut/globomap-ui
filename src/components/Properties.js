/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import './css/Properties.css';

class Properties extends Component {

  constructor(props) {
    super(props);
    this.state = {
      showMore: false
    }
    this.buildProperties = this.buildProperties.bind(this);
  }

  render() {
    return (
      <div className="item-properties">
      {this.buildProperties(this.props.item)}
      </div>
    );
  }

  buildProperties(item) {
    let properties = item.properties || {};
    let propertiesMetadata = item.propertiesMetadata;
    let convertedDate, formattedDate, props;

    convertedDate = new Date(parseInt(item.timestamp, 10) * 1000);
    formattedDate = convertedDate.toLocaleString('pt-BR');

    props = Object.keys(properties).map((key, index) => {
      let val = properties[key];

      if(typeof val === 'boolean') {
        val = val ? 'yes' : 'no';
      }

      if(val instanceof Object) {
        let initial = [],
            remaining = [];

        Object.keys(val).forEach((objectKey, i) => {
          i < 5
            ? initial.push(<span key={i}>{objectKey}: {val[objectKey]}</span>)
            : remaining.push(<span key={i}>{objectKey}: {val[objectKey]}</span>);
        });

        val = <div>
                <div className="prop-initial">{initial}</div>
                {remaining.length > 0 &&
                  <button className="btn-show-more topcoat-button--quiet"
                    onClick={(e) => this.setState({ showMore: !this.state.showMore })}>
                      show {!this.state.showMore ? 'more' : 'less'}
                  </button>}
                {this.state.showMore &&
                  <div className="prop-remaining">{remaining}</div>}
              </div>;
      }

      if(val instanceof Array) {
        val = <div>{properties[key].map(o => <span key={o}>{o}</span>)}</div>;
      }

      return <tr key={key}>
               <th>{(propertiesMetadata &&
                  propertiesMetadata[key].description) || key}</th>
               <td>{val}</td>
             </tr>;
    });

    props.push(
      <tr key="timestamp">
        <th>timestamp</th>
        <td>{formattedDate}</td>
      </tr>
    )

    this.props.hasId &&
      props.push(<tr key="id">
                  <th>id</th>
                  <td>{item.id}</td>
                 </tr>)

    return <table>
              {props.length > 0 &&
                <tbody>{props}</tbody>}
           </table>;
  }

}

export default Properties;
