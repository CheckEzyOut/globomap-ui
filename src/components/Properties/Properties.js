/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Properties.css';

const DATE_FORMAT_LANGUAGE = 'pt-BR';

class Properties extends Component {

  constructor(props) {
    super(props);
    this.state = {
      showMore: false,
      subProps: []
    }

    this.buildProperties = this.buildProperties.bind(this);
    this.buildProps = this.buildProps.bind(this);
    this.toggleProps = this.toggleProps.bind(this);
  }

  toggleProps(event, item) {
    event.preventDefault();
    let sub = this.state.subProps.slice();

    if (sub.includes(item)) {
      sub.splice(sub.indexOf(item), 1);
    } else {
      sub.push(item);
    }

    this.setState({ subProps: sub });
  }

  buildProps(obj, level) {
    return Object.keys(obj).map((key, index) => {
      let value = obj[key];
      let itemKey = key + level;
      let isObject = (value instanceof Object);

      if (isObject) {
        return (
          <div key={key} className={'l-'+level}>
            <div className="item-prop">
              <span>{key}</span>
              <span>
                <a href="#sup-props" onClick={(e) => this.toggleProps(e, itemKey)}>
                  {!this.state.subProps.includes(itemKey)
                    ? <span><i className="icon fa fa-caret-right"></i> show</span>
                    : <span><i className="icon fa fa-caret-down"></i> close</span>}
                </a>
              </span>
            </div>
            {this.state.subProps.includes(itemKey) &&
              <div className="sub-item-prop">
                {this.buildProps(value, level+1)}
              </div>}
          </div>
        );
      }

      if (typeof value === 'boolean') {
        value = value ? 'yes' : 'no';
      } else if (value instanceof Array) {
        value = (
          <div>
            {value.map(o => <span key={o}>{o}</span>)}
          </div>
        );
      }

      return (
        <div key={key} className={'l-'+level}>
          <div className="item-prop">
            <span>{key}</span>
            <span>{value}</span>
          </div>
        </div>
      );
    });
  }

  buildProperties(item) {
    const properties = item.properties || {};
    let content = this.buildProps(properties, 0);

    console.log(item.timestamp);

    const itemTimestamp = new Date(parseInt(item.timestamp, 10) * 1000);
    content.push(<div key="timestamp" className="item-prop">
                   <span>timestamp</span>
                   <span>{itemTimestamp.toLocaleString(DATE_FORMAT_LANGUAGE)}</span>
                 </div>);

    this.props.hasId &&
      content.push(<div key="id" className="item-prop">
                     <span>id</span>
                     <span>{item.id}</span>
                   </div>);

    return content;
  }


  render() {
    return (
      <div className="item-properties">
        {this.buildProperties(this.props.item)}
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {
    hasId: state.app.hasId
  };
}

export default connect(
  mapStateToProps,
  null
)(Properties);
