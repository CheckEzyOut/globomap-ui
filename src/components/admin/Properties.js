/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import update from "immutability-helper";
import './css/Properties.css';
import { TextInput } from "./controls/Text";
import { FormCloseButton, FormPlusButton } from "./controls/Button";
import { PlusIcon, CloseIcon } from "./controls/icon";

class Properties extends Component {

  updateChartData(newData) {
    this.props.setStateNode(update(this.props.node, {data: newData}));
  }

  _changeCommon = (event) => {
    let key = event.target.attributes.name.nodeValue;
    let value = event.target.value;
    this.setState({
      [key]: value
    })
    // this.updateChartData({[key]: {$set: value}});
  }

  _handlePointAdd = () => {
    // let serieKey = this.state.serieKey + this.serieKeyInterval;
    // this.setState({serieKey});

    let newItemData = {name: "", value: ""};
    this.updateChartData({$push: [newItemData]});
  }

  renderData() {
    return this.props.node.data.map(function(data, index) {
      return (
        <div key={"key-value" + index} className="key-value">
          <TextInput
            key="key"
            name="key"
            className="key"
            placeholder="Chave"
            onChange={this._changeCommon}
            defaultValue={data.name} />
          <TextInput
            key="value"
            name="value"
            className="value"
            placeholder="Valor"
            onChange={this._changeCommon}
            defaultValue={data.value} />
        </div>
      )
    }, this)
  }

  render() {
    return (
      <div className="data">
        {this.renderData()}
        <div className="add">
          <FormPlusButton
            name="handlePointAdd"
            onClick={this._handlePointAdd}>
            <PlusIcon /> adicionar
          </FormPlusButton>
        </div>
      </div>
    )
  }
}

export default Properties;
