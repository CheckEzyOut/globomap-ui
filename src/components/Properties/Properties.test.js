import React from 'react';
import { shallow } from 'enzyme';
import { Properties } from './Properties';
import { node } from '../__fixtures__';

describe("Properties", function() {
  it('should render Properties component', () => {
    const component = shallow(<Properties item={node} />);
    expect(component).toMatchSnapshot();
  });
});
