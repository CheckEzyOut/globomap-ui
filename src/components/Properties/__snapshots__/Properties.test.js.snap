// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties should render Properties component 1`] = `
<div
  className="item-properties"
>
  <div
    className="l-0"
    key="equipment_type"
  >
    <div
      className="item-prop"
    >
      <span>
        equipment_type
      </span>
      <span>
        Servidor Virtual
      </span>
    </div>
  </div>
  <div
    className="l-0"
    key="host"
  >
    <div
      className="item-prop"
    >
      <span>
        host
      </span>
      <span>
        cmal18mp11lc03
      </span>
    </div>
  </div>
  <div
    className="l-0"
    key="ips"
  >
    <div
      className="item-prop"
    >
      <span>
        ips
      </span>
      <span>
        <a
          href="#sup-props"
          onClick={[Function]}
        >
          <span>
            <i
              className="icon fa fa-caret-right"
            />
             show
          </span>
        </a>
      </span>
    </div>
  </div>
  <div
    className="l-0"
    key="uuid"
  >
    <div
      className="item-prop"
    >
      <span>
        uuid
      </span>
      <span>
        5d9ed63b-1985-4d3a-98d4-8b0195ce1c53
      </span>
    </div>
  </div>
  <div
    className="item-prop"
    key="timestamp"
  >
    <span>
      timestamp
    </span>
    <span>
      2017-11-23 10:44:51
    </span>
  </div>
  <div
    className="item-prop"
    key="id"
  >
    <span>
      id
    </span>
    <span>
      5d9ed63b-1985-4d3a-98d4-8b0195ce1c53
    </span>
  </div>
</div>
`;
